src/main/java/fruit/shop/strategy/ActivityStrategyImpl.java

```diff
@@ -0,0 +1,24 @@
+package fruit.shop.strategy;
+
+public class ActivityStrategyImpl implements ActivityStrategy {
+    private static final String BALANCE = "b";
+    private static final String SUPPLY = "s";
+    private static final String PURCHASE = "p";
+    private static final String RETURN = "r";
+
+    @Override
+    public int setValueAccordingToOption(String option, int currentValue, int value) {

```

Comment:
what you receive is basically a transaction. Instead of switch case you could use a map (Operation -> Handler). Weird naming, go for something like `processTransaction`

Reply:
Don't understand properly what you meant when you said to use map instead of switch case.

Reply:
Don't understand properly what you meant when you said to use map instead of switch case.